carma-messenger-core/cpp_message/src/SDSM_Message.cpp

```diff
@@ -0,0 +1,1067 @@
+            }
+            else if(temp_lat > j2735_v2x_msgs::msg::Position3D::LATITUDE_MAX){
+
+                RCLCPP_WARN_STREAM(node_logging_->get_logger(),"Encoding latitude value greater than max, setting to max");
+                temp_lat = j2735_v2x_msgs::msg::Position3D::LATITUDE_MAX;
+            }
+            message->value.choice.SensorDataSharingMessage.refPos.lat = temp_lat;  
+        }
+        
+        // Position 3D | Long - ref_pos.longitude
+        if(!plainMessage.ref_pos.longitude || plainMessage.ref_pos.longitude == j2735_v2x_msgs::msg::Position3D::LONGITUDE_UNAVAILABLE){
+            message->value.choice.SensorDataSharingMessage.refPos.Long = j2735_v2x_msgs::msg::Position3D::LONGITUDE_UNAVAILABLE;
+        }
+        else{
+            long temp_long = plainMessage.ref_pos.longitude;
+            if(temp_long < j2735_v2x_msgs::msg::Position3D::LONGITUDE_MIN){
+                RCLCPP_WARN_STREAM(node_logging_->get_logger(),"Encoding longitude value less than min, setting to min");
+                temp_long = j2735_v2x_msgs::msg::Position3D::LONGITUDE_MIN;
+            }
+            else if(temp_long > j2735_v2x_msgs::msg::Position3D::LONGITUDE_MAX){
+                RCLCPP_WARN_STREAM(node_logging_->get_logger(),"Encoding longitude value greater than max, setting to max");
+                temp_long = j2735_v2x_msgs::msg::Position3D::LONGITUDE_MAX;
+            }
+            
+            message->value.choice.SensorDataSharingMessage.refPos.Long = temp_long;
+        }
+        
+        // Position3D | *elevation - ref_pos.elevation
+        auto elevation_ptr = create_store_shared<DSRC_Elevation_t>(shared_ptrs);
+
+        if(!plainMessage.ref_pos.elevation_exists || plainMessage.ref_pos.elevation == j2735_v2x_msgs::msg::Position3D::ELEVATION_UNAVAILABLE){
+            *elevation_ptr = j2735_v2x_msgs::msg::Position3D::ELEVATION_UNAVAILABLE;
+        }
+        else{
+            long temp_elevation = plainMessage.ref_pos.elevation;
+            if(temp_elevation < j2735_v2x_msgs::msg::Position3D::ELEVATION_MIN){
+                RCLCPP_WARN_STREAM(node_logging_->get_logger(),"Encoding elevation value less than min, setting to min");
+                temp_elevation = j2735_v2x_msgs::msg::Position3D::ELEVATION_MIN;
+            }
+            else if (temp_elevation > j2735_v2x_msgs::msg::Position3D::ELEVATION_MAX){
+                RCLCPP_WARN_STREAM(node_logging_->get_logger(),"Encoding elevation value greater than max, setting to max");
+                temp_elevation = j2735_v2x_msgs::msg::Position3D::ELEVATION_MAX;
+            }
+            *elevation_ptr = temp_elevation;
+            
+        }
+        message->value.choice.SensorDataSharingMessage.refPos.elevation = elevation_ptr;
+
+
+        // PositionalAccuracy | semiMajor - semi_minor
+        if(!plainMessage.ref_pos_xy_conf.semi_major || plainMessage.ref_pos_xy_conf.semi_major == j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE){
```

Comment:
This `NOT` check on the int will only check against a zero value. Since 0 is an acceptable min value, i think we can remove that part of the check from both semi-major, semi-minor and orientation

Reply:
Also any reason we're not handling the min in these cases?